src/plugins/devices/configuration-strategies/tcp-strategy.js

```diff
@@ -0,0 +1,58 @@
+const { CGMinerApi } = require('../cgminer-api')
+const { Status } = require('../consts')
+
+const { ConfigurationStrategyInterface } = require('./strategy.interface')
+
+/**
+ * @class
+ * @implements {ConfigurationStrategyInterface}
+ */
+class TcpConfigurationStrategy {
+  constructor(host, abort) {
+    this.host = host
+    this.abort = abort
+  }
+
+  /**
+   * @returns {Promise<Boolean>}
+   */
+  async isAvailable() {
+    const api = new CGMinerApi()
+    await api.connect({ host: this.host, abort: this.abort })
+    return api.hasPrivilegedAccess()
+  }
+
+  /**
+   * Adds new pool to configuration and set highest priority
+   *
+   * @param {String} pool
+   * @param {String} poolUser
+   * @returns {Promise<boolean>} Returns true if successfully updated configuration
+   */
+  async setPool(pool, poolUser) {
+    try {
+      const api = new CGMinerApi()
+      await api.connect({ host: this.host, abort: this.abort })
+
+      const result = await api.addPool(pool, poolUser)
+      const status = result.STATUS[0].STATUS
```

Comment:
if result is undefined this is going to fail

please consider throwing error inside `api.addPool`

Reply:
i would even consider moving the parsing logic inside api

therefore the method would return Promise<void> or rejection

and that is I think a preferable way to return results from success/error functions

Reply:
updated